web/server/codechecker_server/cmd/server.py

```diff
@@ -499,15 +508,20 @@ def check_product_db_status(cfg_sql_server, migration_root, environ):
                                                        interactive=False,
                                                        env=environ)
         db_location = db.get_db_location()
-        ret = db.connect()
-        s_ver = db.get_schema_version()
-        if s_ver in db_errors:
-            s_ver = None
-        prod_status[pd.endpoint] = (ret, s_ver, package_schema, db_location)
 
-    sess.commit()
-    sess.close()
-    engine.dispose()
+        try:
+            status = db.connect()
+            s_ver = db.get_schema_version()
+            if s_ver in db_errors:
+                s_ver = None
+            prod_status[pd.endpoint] = (status, s_ver, package_schema,
+                                        db_location)
+        except Exception:
```

Comment:
Was there a concrete exception occuring here? Shouldn't we expect for that?

Reply:
The main exception here was from the absolutely unnecessary `sess.commit()` **timing out**. After removing that line from here and moving it before the products are iterated, I was not able to produce an exception here. However, the entire server, in principle, should NOT die just because one product had an issue, so I think a catch-all case is appropriate here. Keeping the server alive should be paramount.